implementation/pubsub/rabbitmq/common/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/common/RabbitConnection.java

```diff
@@ -93,12 +99,25 @@ public void setConfig(Dictionary<String, String> config) {
     public void connect() throws IOException {
         // Create a channel if not present
         if (!isConnected()) {
-            try {
-                connection = connectionFactory.newConnection();
-                getLogger().info("Connected to RabbitMQ");
-            } catch (ConnectException | TimeoutException e) {
-                getLogger().error("Could not connect to RabbitMQ: {}", e.getMessage());
-                throw new IOException(e);
+            int attempt = 0;
+
+            while (connection == null) {
+                attempt++;
+
+                try {
+                    connection = connectionFactory.newConnection();
+                    getLogger().info("Connected to RabbitMQ");
+                } catch (ConnectException | TimeoutException e) {
+                    getLogger().error("Could not connect to RabbitMQ (attempt {}): {}", attempt, e.getMessage());
+
+                    if (attempt >= MAX_CONNECTION_ATTEMPTS) {
+                        throw new IOException(e);
+                    } else {
+                        try {
+                            Thread.sleep(CONNECTION_ATTEMPT_TIMEOUT);
+                        } catch (InterruptedException ignored) {}
```

Comment:
Throw new exception as service might have been stopped. Or just do not catch. Right now, service cannot be stopped as thread may not be interrupted